lib/internal/crypto/webidl.js

```diff
@@ -165,7 +166,16 @@ converters.object = (V, opts) => {
 };
 
 function isNonSharedArrayBuffer(V) {
-  return ObjectPrototypeIsPrototypeOf(ArrayBufferPrototype, V);
+  if (ObjectPrototypeIsPrototypeOf.call(ArrayBufferPrototype, V)) {
```

Comment:
I think we should use the `isArrayBuffer` function from `internal/util/types`.

```
  return isArrayBuffer(V);
```

It is documented at https://nodejs.org/api/util.html#utiltypesisarraybuffervalue, and recognize ArrayBuffers from a different realm.

Reply:
+1. also, this isn't a correct change; primordials are call-bound, so you don't need `.call`.